distribution/lib/Standard/Database/0.0.0-dev/src/Internal/Upload/Operations/Internal_Core.enso

```diff
@@ -128,3 +130,11 @@ internal_upload_database_table (source_table : DB_Table) connection table_name p
 check_outside_transaction =
     if In_Transaction.is_in_transaction then
         Panic.throw (Illegal_State.Error "Preparing Table_Upload_Operation should itself be called outside of transaction. This is a bug in the Database library.")
+
+## PRIVATE
+resolve_temp_table_name connection:Connection temporary:Boolean table_name:Text -> Text =
+    case temporary of
+        False -> table_name
+        True -> case connection.dialect.temp_table_style of
+            Temp_Table_Style.Temporary_Table -> table_name
+            Temp_Table_Style.Hash_Prefix -> "#" + table_name
```

Comment:
Shall we also have a check that throws an error if `(temporary == False) && (table_name.starts_with "#")`?